internal/exec/engine.go

```diff
@@ -76,7 +77,7 @@ func (e Engine) Run(stageName string) bool {
 
 	e.Logger.PushPrefix(stageName)
 	defer e.Logger.PopPrefix()
-	return stages.Get(stageName).Create(e.Logger, &e.client, e.Root).Run(config.Append(baseConfig, config.Append(e.OemBaseConfig, cfg)))
+	return stages.Get(stageName).Create(e.Logger, &e.client, e.Root, e.OEM).Run(config.Append(baseConfig, config.Append(e.OemBaseConfig, cfg)))
```

Comment:
This bit is necessary for enabling IAM role credential fetching on EC2. The code in `internal/resource/url.go` could just always attempt to fetch IAM credentials instead of enabling that on a specific platform, but @bgilbert and I agreed that solution isn't acceptable. I could make a `StageCreatorState` struct or some such to pass into `Create` if you think that'd make it easier to manage all the bits necessary here, or alternatively if you have a better idea of how to make the oem string available inside the stages I'd be happy to implement it.